server/py/services/api/crud/model_monitoring/deployment.py

```diff
@@ -290,8 +295,10 @@ def apply_and_create_stream_trigger(
         """
 
         # Get the stream path from the configuration
-        stream_path = services.api.crud.model_monitoring.get_stream_path(
-            project=self.project, function_name=function_name
+        stream_path = mlrun.model_monitoring.get_stream_path(
+            project=self.project,
+            function_name=function_name,
+            secret_provider=self._secret_provider,
```

Comment:
I'm not sure if this is the relevant PR, but in the near future, we need to return the full profile to include the access_key or SSL (kafka).